tests/data_context/cloud_data_context/test_datasource_crud.py

```diff
@@ -242,3 +242,113 @@ def json(self):
         # Make sure the name is populated correctly into the created datasource
         assert retrieved_datasource.name == datasource_name
         assert retrieved_datasource.config["name"] == datasource_name
+
+
+@pytest.mark.cloud
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "config_includes_name_setting",
+    [
+        pytest.param("name_supplied_separately", id="name supplied separately"),
+        pytest.param("config_includes_name", id="config includes name"),
+        pytest.param(
+            "name_supplied_separately_and_included_in_config",
+            id="name supplied separately and config includes name",
+            marks=pytest.mark.xfail(strict=True, raises=TypeError),
+        ),
+    ],
+)
+def test_data_context_in_cloud_mode_add_datasource(
+    config_includes_name_setting: str,
+    empty_cloud_data_context: CloudDataContext,
+    datasource_config: DatasourceConfig,
+    datasource_name: str,
+    ge_cloud_base_url: str,
+    ge_cloud_organization_id: str,
+    request_headers: dict,
+):
+    """A CloudDataContext should save to the cloud backed Datasource store when calling add_datasource. When saving, it should use the id from the response
+    to create the datasource."""
+
+    context: CloudDataContext = empty_cloud_data_context
+    # Make sure the fixture has the right configuration
+    assert isinstance(context, CloudDataContext)
+    assert isinstance(
+        context._datasource_store._store_backend,
+        GeCloudStoreBackend,
+    )
+    assert context.ge_cloud_mode
+    assert len(context.list_datasources()) == 0
+
+    # Setup
+    datasource_id: str = "some_uuid"
+    datasource_config_with_name: DatasourceConfig = copy.deepcopy(datasource_config)
+    datasource_config_with_name.name = datasource_name
+
+    def mocked_post_response(*args, **kwargs):
+        class MockResponse:
+            def __init__(self, json_data: dict, status_code: int) -> None:
+                self._json_data = json_data
+                self._status_code = status_code
+
+            def json(self):
+                return self._json_data
```

Comment:
Any value in bringing this into a global fixture? Perhaps we can leave this as is for now but promote if it has more applicability elsewhere?

Reply:
I think that is a good plan - plenty of refactor opportunities with these tests!